### PR TITLE
Joda-time removal in favor of Java 8 Date/Time API

### LINKED ISSUE #616: Migrate to Java 8 Date Time API (instead of Joda-time)
**Why?**
Joda-time is not necessary when library is already compiled for Java 8. Moreover exposing Joda-time on API (like AbsoluteDateFilter) force consumers of this library to import Joda-time.

**How?**
Migrate to Java 8 Date Time API in whole Gooddata-java, remove Joda-time from dependencies and release it as next major version

### PR DESCRIPTION
**Pull request by To-da**
closes #616

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

File: gooddata-java-model/pom.xml [hunk 1]
```diff
@@ -14,16 +14,6 @@
         <dependency>
             <groupId>com.gooddata</groupId>
             <artifactId>gooddata-rest-common</artifactId>
-            <exclusions>
```
Comment (jimirocks) [inline gooddata-java-model/pom.xml]: hmm are you sure about this?

File: gooddata-java-model/src/main/java/com/gooddata/sdk/model/warehouse/Warehouse.java [hunk 2]
```diff
@@ -5,16 +5,19 @@
  */
 package com.gooddata.sdk.model.warehouse;
 
-import com.fasterxml.jackson.annotation.*;
-import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
-import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import com.fasterxml.jackson.annotation.JsonCreator;
```
Comment (jimirocks) [inline gooddata-java-model/src/main/java/com/gooddata/sdk/model/warehouse/Warehouse.java]: The `annotation.*` import is pretty repeating pattern - looks like someone made it by intention. @liry do you have any opinion on this?

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-java-merge-compliance-check/119/ : SUCCESS in 22s

**Pull request merged**